### ISSUE TITLE
last death time variable or last respawn time variable.

### ISSUE THREAD

**Issue by Mulliganisoup** [Xian55/WowClassicGrindBot#529]
**Is your feature request related to a problem? Please describe.**
I am looking for a variable to add to the class config .JSON files that I can use to base logic off of the last time you respawned/died.
**Describe the solution you'd like**
I found 
( Core.CombatLog.DeadGuid	
  Core.AddonBits.Dead )
In the raw values page but I was unable to find the index values for them to create a variable. 
**Describe alternatives you've considered**
I tried using death count but that just creates an infinite loop. 
**Additional context**
Basically this is for when the bot respawn's near mobs and starts instantly attacking them. 
It could also be used for scrolls and buffs that you want to use before reentering combat.

**Comment by Xian55** [comment]
Hey, thanks for reaching out.

> Basically this is for when the bot respawn's near mobs and starts instantly attacking them.
It could also be used for scrolls and buffs that you want to use before reentering combat.

Sometimes you can't prevent entering combat right after you respawn! Just after respawn, you are already at 50% health and mana, buffing right away might not be always the best idea, using 1 or 2 GCD can cost your life.

Let me enlighten you about another way to solve your issue without adding any extra code.

You can adjust the [KeyAction.Cost(default is 18)](https://github.com/Xian55/WowClassicGrindBot/blob/af927eda7e9cef7098df87f058ce3a4cf1668eea/Core/ClassConfig/KeyAction.cs#L14) value for AdhocGoals(buffs) to have higher priority.

**Lower** the value means higher priority.

By default the [AdhocGoal / buff actions relies on the KeyAction.Cost](https://github.com/Xian55/WowClassicGrindBot/blob/af927eda7e9cef7098df87f058ce3a4cf1668eea/Core/Goals/AdhocGoal.cs#L8). You can put it below [CombatGoal which has Cost of 4](https://github.com/Xian55/WowClassicGrindBot/blob/af927eda7e9cef7098df87f058ce3a4cf1668eea/Core/Goals/CombatGoal.cs#L12).

This `GoapGoal.Cost` or `KeyAction.Cost` values are visible on the BlazorServer. Just before the name
![image](https://github.com/Xian55/WowClassicGrindBot/assets/367101/e62f8972-761f-4b0e-a5b1-a008824ed3ee)


Here's an example of Warlock [Life Tap](https://github.com/Xian55/WowClassicGrindBot/blob/af927eda7e9cef7098df87f058ce3a4cf1668eea/Json/class/Warlock_66_Demo_pet_pull.json#L273) ability or the Warrior [Battle Shout](https://github.com/Xian55/WowClassicGrindBot/blob/af927eda7e9cef7098df87f058ce3a4cf1668eea/Json/class/Warrior_63_Arms.json#L254) or Death Knight [Horn of Winter](https://github.com/Xian55/WowClassicGrindBot/blob/af927eda7e9cef7098df87f058ce3a4cf1668eea/Json/class/DeathKnight_80_Unholy.json#L220)

By default `AdhocGoal` based `KeyAction(s)` wont get executed during combat, you can bypass that with `"InCombat": "i don't care"` which means no matter if the player is in combat or not.
